src/api/spec/factories/staging_workflow.rb

```diff
@@ -20,9 +20,42 @@
         letters = [*'A'..'Z']
         new_staging_projects_count.times do |index|
           letter = letters[index + staging_workflow.staging_projects.count]
-          staging_workflow.staging_projects << create(:staging_project, name: "#{project.name}:Staging:#{letter}")
+          staging_workflow.staging_projects << create(:staging_project, name: "#{staging_workflow.project.name}:Staging:#{letter}", maintainer: staging_workflow.managers_group)
+        end
+      end
+
+      factory :staging_workflow_with_sources do
+        transient do
+          bs_request_count { 10 }
+        end
+
+        after(:create) do |staging_workflow, evaluator|
+          target_packages = create_list(:package, evaluator.bs_request_count, project: staging_workflow.project)
+          source_project = create(:project)
+          target_packages.each do |package|
+            create(:package_with_file, name: package.name, project: source_project)
+          end
+          target_packages.each do |target_package|
+            create(:bs_request_with_submit_action, target_project: staging_workflow.project,
```

Comment:
depending on your goals here, you may want to spam some reviews in the requests. The way you have it, all projects are building/acceptable. Most likely your packages will end being `excluded` as they have no spec file.

So throwing some reviews and checks into the projects will make it more colourful.